src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -178,13 +179,77 @@ SettingsLoader::SettingsLoader(const std::string_view& userJSON, const std::stri
 // (meaning profiles specified by the application rather by the user).
 void SettingsLoader::GenerateProfiles()
 {
-    _executeGenerator(PowershellCoreProfileGenerator{});
-    _executeGenerator(WslDistroGenerator{});
-    _executeGenerator(AzureCloudShellGenerator{});
-    _executeGenerator(VisualStudioGenerator{});
+    PowershellCoreProfileGenerator powerShellGenerator{};
+    _executeGenerator(powerShellGenerator);
+
+    WslDistroGenerator wslGenerator{};
+    _executeGenerator(wslGenerator);
+
+    AzureCloudShellGenerator acsGenerator{};
+    _executeGenerator(acsGenerator);
+
+    VisualStudioGenerator vsGenerator{};
+    _executeGenerator(vsGenerator);

```

Comment:
Is it really necessary to extend the lifetime of these generators? They only get destroyed at the end of the scope now.

Reply:
What motivated this change was that the `PowershellCoreProfileGenerator` already does all the work of finding the existing PowerShell instances; it uses that information to generate (and configure) the appropriate amount of PowerShell profiles.

We need an instance of the profile generator lying around so that we can reuse that information. That way, if we detect that there's an instance of PowerShell installed, we hide the installer stub.

Admittedly, it's not ideal, but its all in one function. Any ideas on how to get around it? Is it worth making the PowershellCoreProfileGenerator special? Or maybe I should pull out the `GetPowerShellInstances()` code into a static function and undo the work of keeping the generators around (I'll give that a try).

Reply:
I would make `PowershellCoreProfileGenerator` special or introduce scopes for each of them. `VisualStudioGenerator` in particular does *a lot* of COM work and I think it's worth destroying that as soon as possible.

Reply:
Fair. Just learned about [templated lambda functions](https://stackoverflow.com/questions/3575901/can-lambda-functions-be-templated) and this looks _so nice_ now!